### PR TITLE
Re-Add: Controller-Manager controllers: Set reconcile success condition

### PR DESCRIPTION
**Pull request by alvaroaleman**
…concile success condition (#4519)" (#4529)"

This reverts commit 7dd565ff5af3dee9eb280d2682bc658b4e1aecef.

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
none
```

### CODE REVIEW

File: api/pkg/controller/monitoring/monitoring_controller.go [hunk 1]
```diff
@@ -217,10 +221,10 @@ func (r *Reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 
 	if err := controllerutil.SetSeedResourcesUpToDateCondition(ctx, cluster, r, successfullyReconciled); err != nil {
```
Comment (xrstf) [inline api/pkg/controller/monitoring/monitoring_controller.go]: This is now being called for *every* cluster, even paused and foreign worker-name clusters.

Comment (alvaroaleman) [reply api/pkg/controller/monitoring/monitoring_controller.go]: Hmm that is indeed wrong. But it doesn't explain why this change fails tests, only how tests would interrupt each other after it was merged?

**Comment by alvaroaleman** [comment]
@xrstf PTAL, no conflicts whatsoever :)

File: api/pkg/controller/monitoring/monitoring_controller.go (hunk 1 above)
Comment (alvaroaleman) [reply api/pkg/controller/monitoring/monitoring_controller.go]: Fixed

Comment (alvaroaleman) [reply api/pkg/controller/monitoring/monitoring_controller.go]: This is pretty ugly, I want to do a follow-up that moves all of the `SeedResourcesUpToDate` handling into a distinct controller

**Pull request merged**